sim/midas/src/main/scala/midas/core/StreamEngine.scala

```diff
@@ -0,0 +1,78 @@
+// See LICENSE for license details.
+
+package midas.core
+
+import chisel3._
+import chisel3.util._
+import chisel3.experimental.IO
+import freechips.rocketchip.amba.axi4._
+import freechips.rocketchip.config.{Parameters, Field}
+import freechips.rocketchip.diplomacy._
+
+import midas.widgets._
+
+/**
+  * Set by a platform config to instantiate the supported StreamEnginer for
+  * that host.  e.g. F1 uses CPU-driven XDMA and so uses an engine that only
+  * uses the AXI4M interface.
+  */
+case object StreamEngineInstantiatorKey extends Field[(StreamEngineParameters, Parameters) => StreamEngine](
+  (e: StreamEngineParameters, p: Parameters) => new CPUManagedStreamEngine(p, e))
+
+
+/**
+  * Parameters that define a stream that are defined by bridges and collected
+  * by the engine.
+  */
+trait StreamParameters {
+  def name: String
+  def fpgaBufferDepth: Int
```

Comment:
Perhaps a good name for this that might be implementation agnostic: `minimumQueueCapacity` which could then be set to the batch size the driver expects to use.

Reply:
better: desiredBatchDepth (in 512b chunks) or desiredBatchBytes? 

Unfortunately good choices for these values probably depend on the transport / Stream implementation

Reply:
To be clear, this could also mean the host CPU queue/circular-buffer size? Like in the U250 case this would be the designated size of the circular buffer?

Reply:
If so, I tend to prefer the `batchBytes` version.

Reply:
I was thinking the circular buffer would be much larger (since they are relatively inexpensive). I think we agreed to punt on the names until we had the PCI-M based implementation in front of us.